workflow-controller/controller-service/src/main/java/io/hyscale/controller/commands/HyscaleGenerateServiceManifestsCommand.java

```diff
@@ -89,15 +99,27 @@ public Integer call() throws Exception {
         if (!CommandUtil.isInputValid(this)) {
             return ToolConstants.INVALID_INPUT_ERROR_CODE;
         }
+        
+        Map<String, String> serviceProfileMap = new HashMap<String, String>();
+        try {
+            serviceProfileMap = ServiceProfileUtil.getServiceProfileMap(profiles);
+        } catch (HyscaleException e) {
+            WorkflowLogger.error(ControllerActivity.CANNOT_PROCESS_SERVICE_PROFILE, e.getMessage());
+            throw e;
+        }
         boolean isFailed = false;
-        for (int i = 0; i < serviceSpecs.length; i++) {
+        for (String serviceSpecPath : serviceSpecs) {
 
             WorkflowContext workflowContext = new WorkflowContext();
-            String serviceName = null;
+            String serviceName = ServiceProfileUtil.getServiceName(serviceSpecPath);
+            String profilePath = serviceProfileMap.remove(serviceName);
             try {
-                File serviceSpecFile = new File(serviceSpecs[i]);
-                ServiceSpec serviceSpec = serviceSpecMapper.from(serviceSpecFile);
-                serviceName = serviceSpec.get(HyscaleSpecFields.name, String.class);
+                File serviceSpecFile = new File(serviceSpecPath);
+                File profileFile = null;
+                if (profilePath != null) {
+                    profileFile = new File(profilePath);
+                }
+                ServiceSpec serviceSpec = serviceSpecMapper.from(serviceSpecFile, profileFile);
```

Comment:
Duplicate code in generation & deploy. Going on it will be difficult to maintain